packages/block-library/src/latest-posts/editor.scss

```diff
@@ -14,10 +14,6 @@
 	display: inline;
 }
 
-.edit-post-visual-editor .wp-block-latest-posts.is-grid li {
-	margin-bottom: 20px;
-}
```

Comment:
This style was applied in my testing in non-iframe mode but there's a very similar style in the style.scss which for me means that removing it is fine.

You can test by using the latest posts block in the post editor, switch to "grid" and enable custom fields (to enable non-iframe mode).